src/lib.rs

```diff
@@ -21,3 +56,14 @@ mod tests {
 pub fn zeros(n: usize) -> Vec<isize> {
     repeat(0).take(n).collect::<Vec<isize>>()
 }
+
+pub struct QArray {
+    dimensions: Vec<usize>,
+    data: Vec<isize>
+}
+
+pub fn index(array: &QArray, coordinates: Vec<usize>) -> usize {
+    coordinates.into_iter().enumerate().map(|(idx, coord)| {
+        array.dimensions.iter().take(idx).fold(coord, |acc, x| acc * x )
+    }).fold(0, |acc, x| acc + x)
```

Comment:
I understand this now, but I doubt I will in 3 weeks time. We might wanna refactor it at some point.